packages/dapp-toolkit/src/modules/wallet-request/crypto/curve25519.ts

```diff
@@ -37,7 +37,7 @@ export const Curve25519: KeyPairProvider = (
     getPublicKey: () => toHex(ed25519.getPublicKey(privateKeyHex)),
     sign: (messageHex: string): Result<string, Error> => {
       try {
-        return ok(toHex(ed25519.sign(privateKeyHex, messageHex)))
+        return ok(toHex(ed25519.sign(messageHex, privateKeyHex)))
```

Comment:
We really should stop using Strings everywhere. I've brought this up in the past.

Best to create a PrivateKey type early. 

And a Exactly32bytes type early as well, and use those, not strings.